bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/handler/AbstractSunSpecHandler.java

```diff
@@ -152,31 +152,58 @@ private void startUp() {
             return;
         }
 
-        Optional<ModelBlock> mainBlock = getAddressFromConfig();
-        if (mainBlock.isPresent()) {
-            publishUniqueAddress(mainBlock.get());
+        // Try properties first
+        @Nullable
+        ModelBlock mainBlock = getAddressFromProperties();
+
+        if (mainBlock == null) {
+            mainBlock = getAddressFromConfig();
+        }
+
+        if (mainBlock != null) {
+            publishUniqueAddress(mainBlock);
             updateStatus(ThingStatus.UNKNOWN);
-            registerPollTask(mainBlock.get());
+            registerPollTask(mainBlock);
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                     "SunSpec item should either have the address and length configuration set or should been created by auto discovery");
             return;
         }
     }
 
+    /**
+     * Load and parse configuration from the properties
+     * These will be set by the auto discovery process
+     */
+    private @Nullable ModelBlock getAddressFromProperties() {
+        Map<String, String> properties = thing.getProperties();
+        if (!properties.containsKey(PROPERTY_BLOCK_ADDRESS) || !properties.containsKey(PROPERTY_BLOCK_LENGTH)) {
+            return null;
+        }
+        try {
+            ModelBlock block = new ModelBlock();
+            block.address = (int) Double.parseDouble(thing.getProperties().get(PROPERTY_BLOCK_ADDRESS));
+            block.length = (int) Double.parseDouble(thing.getProperties().get(PROPERTY_BLOCK_LENGTH));
```

Comment:
Why are you parsing as a double then casting it to an int? Are you expecting the properties to have decimal points?
Could you just parse it as an int instead?

Reply:
Yes of course. I don't know anymore what I was doing here. Possibly a cope&paste mistake.

Reply:
Now I remember what's been the problem here: some older versions of OH return the property value with decimal points even though I set them as integers.

On the current state of 2.5.x branch this didn't happen, but others testing this bundle reported the problem.

I'm not familiar with the internals of OH, and how the properties are stored, or if there has been a change on this recently. @cpmeister do you have any information regarding this?